socketioxide/src/retryer.rs

```diff
@@ -0,0 +1,139 @@
+use crate::errors::RetryerError;
+use engineioxide::{sid_generator::Sid, SendPacket};
+use std::{collections::VecDeque, fmt::Debug};
+use tokio::{sync::mpsc::error::TrySendError, sync::mpsc::Sender};
+
+// todo bin payload and payload should be in one VecDeque,
+// todo sender should be the same for both types of SendPacket
+
+/// The `Retryer` struct represents a retry mechanism for sending packets.
+#[derive(Debug)]
+pub struct Retryer<T: Debug> {
+    sid: Sid,
+    sender: Sender<T>,
+    packet: Option<T>,
```

Comment:
You could remove the generic param, only a `Sender<SendPacket>` is needed, it avoids one more clone

Reply:
if you are sure that we don't use this retryer with another Packet, I'll remove it